client/src/redux/reducers/twitterReducer.js

```diff
@@ -55,55 +60,69 @@ const twitterReducer = (state = initialState, action) => {
     case TWITTER_ADD_STREAM:
       return {
         ...state,
-        streams: [...state.streams, action.payload],
+        streams: [...state.streams, action.payload.id],
+        streamsById: {
+          ...state.streamsById,
+          [action.payload.id]: action.payload,
+        },
+      };
+    case TWITTER_REMOVE_STREAMBYID:
```

Comment:
The `streams` and `streamByIds` should be updated in the same action handler when streams are added / deleted. 

Otherwise, you create call dependencies (action A must be called before action B, otherwise the app might crash). We could talk more about this when we meet next time.

Reply:
see comment below